Радкевич Марсель

SquirrelLeonid opened the pull request:
@lamanoff

SquirrelLeonid committed:
Created a solution

SquirrelLeonid committed:
Added project for console client

SquirrelLeonid committed:
Added project for GUI client

SquirrelLeonid committed:
Import project from TDD task

SquirrelLeonid committed:
Created first version of input words

SquirrelLeonid committed:
Added resx file

SquirrelLeonid committed:
Removed classes from TDD task due to fact that they interfere

SquirrelLeonid committed:
Added InitialWordFormReader

Added IReader interface
Added ProcessExtensions
Added mystem.exe for leading words to initial form

SquirrelLeonid committed:
Removed ProcessExtensions

SquirrelLeonid committed:
Created WordsReading namespace

Moved InitialWordFormReader and IReader to WordsReading namesapce

SquirrelLeonid committed:
Added simplest words filter

SquirrelLeonid committed:
minor restructuring

SquirrelLeonid committed:
Import algorithm from TDD task

SquirrelLeonid committed:
Moved Saving and visuaization modules from tag cloud TDD algorithm to top of project

SquirrelLeonid committed:
Added words frequency analyzer

Added new interfaces to Words module

lamanoff commented on TagCloudContainerTask/TagCloud/Saving/BitmapSaver.cs:
```diff
@@ -0,0 +1,52 @@
+{
+    public class BitmapSaver : IBitmapSaver
+    {
+        private const string BitmapsDirectory = "layouts";
```
А давай это всё принимать в конструкторе? Вместе с типом файла.

SquirrelLeonid replied:
Убрал все статические поля из класса.
Эти данные теперь передаются напрямую в метод

lamanoff commented on TagCloudContainerTask/TagCloud/Saving/BitmapSaver.cs:
```diff
@@ -0,0 +1,52 @@
+        private static readonly ImageFormat ImgFormat = ImageFormat.Png;
+
+        private static readonly string ProjectDirectory
+            = Directory.GetParent(Environment.CurrentDirectory)?.Parent?.Parent?.FullName;
```
А если у нас программа будет лежать просто в папке, то куда он сохранит вывод?)

lamanoff commented on TagCloudContainerTask/TagCloud/Visualization/Drawer.cs:
```diff
@@ -0,0 +1,72 @@
+            Color.Chartreuse
+        };
+
+        public void DrawCanvasBoundary(Graphics g, Size imgSize)
```
Лучше не сокращать переменные, где это можно не делать)

lamanoff commented on TagCloudContainerTask/TagCloud/Visualization/Drawer.cs:
```diff
@@ -0,0 +1,72 @@
+    {
+        private const int LineWidth = 2;
+
+        private readonly List<Color> colors = new List<Color>
```
А давай настройки и цвета тоже принимать в конструкторе.

lamanoff commented on TagCloudContainerTask/TagCloud/Words/Filtering/WordsFilter.cs:
```diff
@@ -0,0 +1,13 @@
+
+namespace TagCloud.Words.Filtering
+{
+    public class WordsFilter : IFilter
```
Если этот фильтр фильтрует по длине, то тогда его следует назвать соответственно.

lamanoff commented on TagCloudContainerTask/TagCloud/Words/Filtering/WordsFilter.cs:
```diff
@@ -0,0 +1,13 @@
+    {
+        public IEnumerable<string> FilterWords(IEnumerable<string> words)
+        {
+            return words.Where(word => word.Length > 3);
```
Может количество символов принимать как параметр? Это всё создаёт дополнительные точки расширения.

lamanoff commented on TagCloudContainerTask/TagCloud/Words/Preprocessing/WordsPreprocessor.cs:
```diff
@@ -0,0 +1,13 @@
+
+namespace TagCloud.WordsPreprocessing
+{
+    public class WordsPreprocessor
```
Этот класс вообще нужен?) Кажется, он просто дублирует `ToLowerCasePreprocessor` и не реализует интерфейс препроцессора.

lamanoff commented on TagCloudContainerTask/TagCloud/Words/Reading/IReader.cs:
```diff
@@ -0,0 +1,9 @@
+{
+    public interface IReader
+    {
+        IEnumerable<string> ReadWordsFromFile(string pathToFile);
```
А если мы захоти считать слова не из файла? Тут получается интерфейс ридера, у которого будет всегда только реализация чтения из файла.

lamanoff commented on TagCloudContainerTask/TagCloud/Words/Reading/InitialWordFormReader.cs:
```diff
@@ -0,0 +1,50 @@
+        private const string DefaultStemArguments = "-nld";
+
+        private readonly string solutionDirectory
+            = Directory.GetParent(Directory.GetCurrentDirectory())?.Parent?.Parent?.FullName;
```
Не стоит завязываться на структуру файлов из солюшна.

lamanoff commented on TagCloudContainerTask/TagCloud/Words/Reading/InitialWordFormReader.cs:
```diff
@@ -0,0 +1,50 @@
+
+namespace TagCloud.Words.Reading
+{
+    public class InitialWordFormReader : IReader
```
Тут две ответственности у класса, который должен выступать только в роде читальщика из файла. Превращение слов в начальную форму нужно внедрять где-то в общий пайплайн обработки. Это своего рода тоже препроцессор.

SquirrelLeonid committed:
Renamed WordsFilter to WordsLengthFilter

SquirrelLeonid committed:
Renamed Graphics g to Graphics graphics in Visualization module

SquirrelLeonid committed:
Added separated FileReader and ConsoleReader.

Also added corresponding interfaces

Removed deprecated IReader interface

SquirrelLeonid committed:
Added writing module

Added encoding parameter to FileReader

SquirrelLeonid committed:
Temporary moved mystem.exe to ./TagCloud/bin/debug

Added encoding parameter to WordsReader
Changes in ToInitFormPreprocessor

SquirrelLeonid committed:
Set "copy to output directory" option for mystem.exe

SquirrelLeonid committed:
Removed deprecated copy of mystem.exe

lamanoff commented on TagCloudContainerTask/TagCloud/Words/Reading/IWordsReader.cs:
```diff
@@ -0,0 +1,11 @@
+{
+    public interface IWordsReader
+    {
+        IEnumerable<string> ReadFromFile(string pathToFile, Encoding encoding);
```
Можно ли как-то унифицировать ридер? Чтобы он принимал, например, стрим, и не был привязан к способу чтения?

lamanoff commented on TagCloudContainerTask/TagCloud/Words/Writing/IWordsWriter.cs:
```diff
@@ -0,0 +1,14 @@
+
+namespace TagCloud.Words.Writing
+{
+    public interface IWordsWriter
```
Также как-то бы унифицировать райтер, например, писать в стрим, который уже будет писать в файл или консоль.

lamanoff commented on TagCloudContainerTask/TagCloud/Words/Writing/ToConsole/ConsoleWriter.cs:
```diff
@@ -0,0 +1,27 @@
+
+        public void WriteToConsole(IEnumerable<string> linesToWrite)
+        {
+            foreach (var item in linesToWrite) Console.WriteLine(item);
```
Инструкцию под циклом следует переносить на новую строку.

SquirrelLeonid committed:
Implemented dirt version of words tagging

Clear Program.Main() body to exclude compile errors

SquirrelLeonid committed:
Implemented universal words reader.

Deleted old version of words reading
Clear Program.cs to exclude complie errors

SquirrelLeonid committed:
Merge branch 'universalReaderFeature'

SquirrelLeonid committed:
Renamed Reading module to Readers

SquirrelLeonid committed:
Implemented universal words writer.

Deleted old version of words writing

Renamed Writing module to Writers

SquirrelLeonid committed:
Implemented words tagging.

Renamed namespaces in Words module

SquirrelLeonid committed:
Restructured TagCloud

->Geometry moved to TagCloud root and renamed to GeometryUtils
-> Algorithms renamed to LayoutAlgorithms
-> interface iLayoter moved to LayoutAlgorithms

first attempt to adapt visualizer to drawing words.

SquirrelLeonid committed:
Change IVisualizer and Interface

SquirrelLeonid committed:
changes in BitmapSaver

-> Class no longer contains static field with file name, file extension and e.t.c.
-> BitmapSaver.Save renamed to BitmapSaver.SaveBitmapTo

SquirrelLeonid committed:
Renamed TagCloud/Saving to TagCloud/Savers

SquirrelLeonid committed:
Created new FileInteractions class library project

It will contain logic that interact with files.

Moved modules Writers, Readers, Savers то FileInteractions project

SquirrelLeonid committed:
Added console argument parsing library to console client

Added static field to Tag

SquirrelLeonid committed:
added using statement to DrawTags method

SquirrelLeonid committed:
Implemented .doc .docx Reader

lamanoff commented on TagCloudContainerTask/FileInteractions/Savers/BitmapSaver.cs:
```diff
@@ -0,0 +1,36 @@
+{
+    public class BitmapSaver : IBitmapSaver
+    {
+        public void SaveBitmapTo(
```
Тоже похоже на статический метод.

SquirrelLeonid replied:
Этот метод идет из интерфейса и не может быть статичным

lamanoff commented on TagCloudContainerTask/TagCloud/Visualization/Drawer.cs:
```diff
@@ -55,18 +57,15 @@ public void DrawCloudBoundary(Graphics graphics, Size imgSize, Point cloudCenter
+            foreach (var tag in tags)
+                graphics.DrawString(
+                    tag.Word,
+                    new Font("times new roman", tag.WordEmSize),
```
Шрифт лучше принимать где-то в конструкторе.

SquirrelLeonid committed:
Restructurization step 1

-> Created project App
-> Moved into that project FileInteractions interfaces and implementations

SquirrelLeonid committed:
Restructurization step 2

-> Moved into App project Words module interfaces and implementations

SquirrelLeonid committed:
Restructurization step 3

-> Moved into App project Visualization and GeometryUtils modules.

SquirrelLeonid committed:
Restructurization step 4

-> Moved into App project LayoutingAlgorithms module

SquirrelLeonid committed:
Added set of  settings holders

SquirrelLeonid committed:
Added fontSettings to use in code

Added MyStem to App project

SquirrelLeonid committed:
Added CloudGenerator

SquirrelLeonid committed:
Added paletteSettings and imageSizeSettings to Drawer

Added imageSizeSettings to CloudGenerator

SquirrelLeonid committed:
Deleted Form1 class

SquirrelLeonid committed:
Added ReaderFactory and layouterFactory

Added set of classes that contains different settings
Added interfaces to infrastructure

SquirrelLeonid committed:
Added helpful infrastructure from di/fractalPainter project

SquirrelLeonid committed:
Added MainForm to GuiClient

Added Microsoft.Extensions.DependencyInjection
Injected DI container

SquirrelLeonid committed:
Added RecreatedImage to some actions

Added new menu that allow redraw image

SquirrelLeonid committed:
Removed deprecated methods from IImageHolder interface

Removed first call of GenerateImage in MainForm constructor

SquirrelLeonid committed:
Hide mystem console window from user

SquirrelLeonid committed:
Added construcor with min word length to WordsLengthFilter

SquirrelLeonid committed:
Organized DI container binding

SquirrelLeonid committed:
Added tests

Removed deprecated files
